src/plugins/plugin.colors.ts

```diff
@@ -0,0 +1,106 @@
+import type {Chart, ChartDataset} from '../types';
+
+export interface ColorsPluginOptions {
+  enabled?: boolean;
+}
+
+type DatasetColorizer = (dataset: ChartDataset, i: number) => void;
+
+interface ColorsDescriptor {
+  backgroundColor?: unknown;
+  borderColor?: unknown;
+}
+
+const BORDER_COLORS = [
+  'rgb(54, 162, 235)', // blue
+  'rgb(255, 99, 132)', // red
+  'rgb(255, 159, 64)', // orange
+  'rgb(255, 205, 86)', // yellow
+  'rgb(75, 192, 192)', // green
+  'rgb(153, 102, 255)', // purple
+  'rgb(201, 203, 207)' // grey
+];
+
+// Border colors with 50% transparency
+const BACKGROUND_COLORS = /* #__PURE__ */ BORDER_COLORS.map(color => color.replace('rgb(', 'rgba(').replace(')', ', 0.5)'));
```

Comment:
might be able to avoid the `__PURE__` part by declaring `BORDER_COLORS` with `as const`

Reply:
@etimberg I checked with @dangreen and he thinks it would make no difference for tree-shaking 🤷